Base/QTCore/qMRMLTranslator.h

```diff
@@ -0,0 +1,16 @@
+#ifndef qMRMLTranslator_h
+#define qMRMLTranslator_h
+
+#include "vtkMRMLTranslator.h"
+
+#include "qSlicerBaseQTCoreExport.h"
+
+class Q_SLICER_BASE_QTCORE_EXPORT qMRMLTranslator: public vtkMRMLTranslator
```

Comment:
Since this class is derived from a VTK base class, it has to be a VTK class, i.e., `vtkMRMLTranslator`.

Reply:
This particular class should probably be called `qSlicerQtVTKTranslationBridge` ?

Note that since its role is to establish a "bridge" so that VTK logic classes can indirectly leverage Qt translation capabilities through polymorphism, it doesn't have to strictly start with the prefix `vtk`.

That said, may be the name `vtkSlicerQtVTKTranslationBridge` would be better ?

Reply:
`vtk` prefix is an important hint about how the class is allowed to be constructed and used. `vtk...` classes are based on `vtkObject` and rely on internal reference counting. So, for clarity and consistency we either use a proper VTK class (with appropriate naming, internal reference counting, private constructor, etc.) or we don't make the translator a VTK class at all. We could make it a Qt class (derived from QObject, following Qt conventions) then, or just make it a simple C++ class (we establishing all rules).

Since initialization and destruction of singleton classes is a difficult topic, I would use already established singleton patterns in Slicer or VTK. The current design seems to be some custom implementation, which may not work correctly in a large application, composed of many shared libraries.